src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -203,8 +203,12 @@ public void FinishQuerying()
 				throw new MySqlException("Unable to connect to any of the specified MySQL hosts.");
 			}
 
-			var socketByteHandler = new SocketByteHandler(m_socket);
-			m_payloadHandler = new StandardPayloadHandler(socketByteHandler);
+#if NETSTANDARD2_0
+ 			var byteHandler = new StreamByteHandler(m_networkStream);
```

Comment:
It would be easier, but there will still be the overhead of allocating `Task`s even to represent those synchronous operations.

I didn't write up the gains (or finish this implementation) but I was finding a performance increase from removing `ValueTask` because its asynchronous version has to allocate a `Task`: https://github.com/bgrainger/MySqlConnector/commits/rewrite-value-task

I'm very interested in testing my experimental code against the new `Socket` and `NetworkStream` in netstandard2.0.